### PR TITLE
Add filter to assest based on source extension not mismatched mime/types

### PR DESCRIPTION
**Pull request by wshafer**

### CODE REVIEW

**Review by RWOverdijk — APPROVED** [review]
I prefer using a mime type that's accurate, especially because the response type should make sense, but if this is what the crowd wants :D :+1:

File: src/Service/AssetFilterManager.php [hunk 1]
```diff
@@ -70,35 +65,60 @@ protected function setConfig($config)
...
+
+        foreach ($filters as $filter) {
+            $this->setFilter($filter, $asset);
+        }
+    }
+
+    /**
+     * Get the filters from config based on path, mimetype or extension.
+     *
+     * @param $path
```
Comment (RWOverdijk) [inline src/Service/AssetFilterManager.php]: Add type

File: src/Service/AssetFilterManager.php [hunk 2]
```diff
@@ -70,35 +65,60 @@ protected function setConfig($config)
...
+     * @param AssetInterface $asset
+     * @return array|mixed
```
Comment (RWOverdijk) [inline src/Service/AssetFilterManager.php]: Newline before return (blocks in docs are also separated)

File: src/Service/AssetFilterManager.php [hunk 3]
```diff
@@ -70,35 +65,60 @@ protected function setConfig($config)
...
+        $extension = strtolower(pathinfo($asset->getSourcePath(), PATHINFO_EXTENSION));
+
+        if (!empty($config[$extension])) {
+            return $config[$extension];
+        }
+
+        return [];
+    }
+
+    protected function setFilter($filter, AssetInterface $asset)
```
Comment (RWOverdijk) [inline src/Service/AssetFilterManager.php]: Add docblock

File: src/Service/AssetFilterManager.php [hunk 4]
```diff
@@ -70,35 +65,60 @@ protected function setConfig($config)
...
+        }
+
+        if (!empty($filter['filter'])) {
+            $this->ensureByFilter($asset, $filter['filter']);
+            return;
         }
+
+        if (!empty($filter['service'])) {
+            $this->ensureByService($asset, $filter['service']);
+            return;
```
Comment (RWOverdijk) [inline src/Service/AssetFilterManager.php]: Newline before return

File: src/Service/AssetFilterManager.php [hunk 5]
```diff
@@ -70,35 +65,60 @@ protected function setConfig($config)
...
+
+    protected function setFilter($filter, AssetInterface $asset)
+    {
+        if (is_null($filter)) {
+            return;
+        }
+
+        if (!empty($filter['filter'])) {
+            $this->ensureByFilter($asset, $filter['filter']);
+            return;
```
Comment (RWOverdijk) [inline src/Service/AssetFilterManager.php]: Newline before return

**Review by RWOverdijk — DISMISSED** [review]

**Comment by RWOverdijk** [comment]
@michaelgooden The hero we need, but don't deserve, or... Anyway, ping.

File: src/Service/AssetFilterManager.php [hunk 6]
```diff
@@ -63,49 +58,84 @@ protected function setConfig($config)
...
+    /**
+     * Set the filter by filter or service
+     *
+     * @param string         $filter
+     * @param AssetInterface $asset
+     */
+    protected function setFilter($filter, AssetInterface $asset)
+    {
+        if (is_null($filter)) {
+            return;
```
Comment (MichaelGooden) [inline src/Service/AssetFilterManager.php:113]: It feels like this should be an error. What is the use case for this "failing" silently?

Comment (wshafer) [reply src/Service/AssetFilterManager.php:113]: I see what you mean. However, this just keeps it's current behavior. All I did was de-nest all the if statements.

Comment (wshafer) [reply src/Service/AssetFilterManager.php:113]: After reading the code... it seems this is here if you set the filters to null in your config (which you would do if you overwrite the global config with local config)...  Like so:

```php
return [
    'AssetManager' => [
        'filters' => [
            'css' => [
                'filter_one' => null
             ],
        ]
    ];
];
```

Everything still works.  If we error out here, we'll have no way to stop using a filter on say a dev box, without replacing all the filters, which may not be what we really want.

**Pull request merged**